libraries/AP_Frsky_Telem/AP_Frsky_SPort_Passthrough.cpp

```diff
@@ -0,0 +1,554 @@
+#include "AP_Frsky_SPort_Passthrough.h"
+
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_BattMonitor/AP_BattMonitor.h>
+#include <AP_GPS/AP_GPS.h>
+#include <AP_HAL/utility/RingBuffer.h>
+#include <AP_InertialSensor/AP_InertialSensor.h>
+#include <AP_Notify/AP_Notify.h>
+#include <AP_RangeFinder/AP_RangeFinder.h>
+#include <GCS_MAVLink/GCS.h>
+
+/*
+for FrSky SPort Passthrough
+*/
+// data bits preparation
+// for parameter data
+#define PARAM_ID_OFFSET             24
+#define PARAM_VALUE_LIMIT           0xFFFFFF
+// for gps status data
+#define GPS_SATS_LIMIT              0xF
+#define GPS_STATUS_LIMIT            0x3
+#define GPS_STATUS_OFFSET           4
+#define GPS_HDOP_OFFSET             6
+#define GPS_ADVSTATUS_OFFSET        14
+#define GPS_ALTMSL_OFFSET           22
+// for battery data
+#define BATT_VOLTAGE_LIMIT          0x1FF
+#define BATT_CURRENT_OFFSET         9
+#define BATT_TOTALMAH_LIMIT         0x7FFF
+#define BATT_TOTALMAH_OFFSET        17
+// for autopilot status data
+#define AP_CONTROL_MODE_LIMIT       0x1F
+#define AP_SIMPLE_OFFSET            5
+#define AP_SSIMPLE_OFFSET           6
+#define AP_FLYING_OFFSET            7
+#define AP_ARMED_OFFSET             8
+#define AP_BATT_FS_OFFSET           9
+#define AP_EKF_FS_OFFSET            10
+#define AP_IMU_TEMP_MIN             19.0f
+#define AP_IMU_TEMP_MAX             82.0f
+#define AP_IMU_TEMP_OFFSET          26
+// for home position related data
+#define HOME_ALT_OFFSET             12
+#define HOME_BEARING_LIMIT          0x7F
+#define HOME_BEARING_OFFSET         25
+// for velocity and yaw data
+#define VELANDYAW_XYVEL_OFFSET      9
+#define VELANDYAW_YAW_LIMIT         0x7FF
+#define VELANDYAW_YAW_OFFSET        17
+// for attitude (roll, pitch) and range data
+#define ATTIANDRNG_ROLL_LIMIT       0x7FF
+#define ATTIANDRNG_PITCH_LIMIT      0x3FF
+#define ATTIANDRNG_PITCH_OFFSET     11
+#define ATTIANDRNG_RNGFND_OFFSET    21
+
+bool AP_Frsky_SPort_Passthrough::init()
+{
+    if (!AP_RCTelemetry::init()) {
+        return false;
+    }
+
```

Comment:
suggested change:
    
// do not initialize serial ports for external data
if (_use_external_data) {
  return true;
}

Reply:
I've done this one in a slightly different way- not calling your parent class' init method isn't usually a good idea.